python/cudf/cudf/core/column/categorical.py

```diff
@@ -47,7 +47,9 @@
     )
 
 
-_DEFAULT_CATEGORICAL_VALUE = -1
+# Using np.int8(-1) to allow silent wrap-around when casting to uint
```

Comment:
I'm guessing we should disallow this cast if this happens, but not for this PR

Reply:
Sorry for not following up.  NumPy allows `np.int8(-1)` to be cast to silently to other integers pretty easily including unsigned, which is what we rely on here.

If NumPy wants to change that, the `_DEFAULT_CATEGORICAL_VALUE` might have to be some mapping/function based on the actual dtype used, but for now this works.